src/luxonis_ml/embeddings/ldf_emb.py

```diff
@@ -0,0 +1,292 @@
+"""
+This script provides utilities for generating embeddings from images, filtering new samples, and inserting them into a Qdrant database.
+
+Modules Used:
+- cv2: For reading and processing images.
+- numpy: For numerical operations.
+- torch: PyTorch library for deep learning.
+- torch.onnx: PyTorch's ONNX utilities.
+- onnx: Open Neural Network Exchange.
+- onnxruntime: Runtime for ONNX models.
+- torchvision: PyTorch's computer vision library.
+- qdrant_client: Client for interacting with Qdrant.
+
+Main Functions:
+- _get_sample_payloads_coco: Extracts payloads from the LuxonisDataset for the COCO dataset format.
+- _get_sample_payloads: Extracts payloads from the LuxonisDataset.
+- _filter_new_samples: Filters out samples that are already in the Qdrant database based on their sample ID.
+- _filter_new_samples_by_id: Filters out samples that are already in the Qdrant database based on their instance ID.
+- _generate_new_embeddings: Generates embeddings for new images using a given ONNX runtime session.
+- _batch_upsert: Performs batch upserts of embeddings to Qdrant.
+- generate_embeddings: Main function that generates embeddings for a given dataset and inserts them into Qdrant.
+
+Constants:
+- DEFAULT_COLLECTION_NAME: Default name for the Qdrant collection.
+- DEFAULT_OUTPUT_LAYER_NAME: Default name for the output layer in the ONNX model.
+- DEFAULT_EMB_BATCH_SIZE: Default batch size for generating embeddings.
+- DEFAULT_QDRANT_BATCH_SIZE: Default batch size for inserting into Qdrant.
+
+Note:
+Ensure that the Qdrant server is running and accessible before using these utilities.
+"""
+
+import cv2
+import numpy as np
+import torch
+import torch.onnx
+import onnx
+import onnxruntime
+import torchvision
+import torchvision.transforms as transforms
+from qdrant_client import QdrantClient
+from qdrant_client.models import Distance, VectorParams, PointStruct, SearchRequest
+from qdrant_client.http import models
+
+DEFAULT_COLLECTION_NAME = "mnist"
+DEFAULT_OUTPUT_LAYER_NAME = "/Flatten_output_0"
+DEFAULT_EMB_BATCH_SIZE = 64
+DEFAULT_QDRANT_BATCH_SIZE = 64
```

Comment:
We likely want to avoid these global variables. Maybe a class definition makes sense here to make these more easily configurable but still easy to use across functions?

Reply:
We could also use a `self.dataset` for LDF, similar to `LuxonisLoader`